### PR TITLE
Techs and bug fixes

### PR DESCRIPTION
**Pull request by Paril**

### CODE REVIEW

File: src/game/default/g_item.c [hunk 1]
```diff
@@ -812,6 +820,160 @@ static void G_UseItem(g_entity_t *ent, g_entity_t *other, g_entity_t *activator)
 }
 
 /**
+ * @brief
+ */
+static g_entity_t *G_TechEntity(const g_tech_t tech) {
+	return G_FindPtr(NULL, LOFS(item), g_media.items.techs[tech]);
+}
+
+
```
Comment (jdolan) [inline src/game/default/g_item.c]: Extra newline here.

File: src/game/default/g_item.c [hunk 2]
```diff
@@ -812,6 +820,160 @@ static void G_UseItem(g_entity_t *ent, g_entity_t *other, g_entity_t *activator)
...
+		if (dist < best_dist) {
+			best_dist = dist;
+		}
+
+		any = true;
+	}
+
+	if (!any) {
+		return Randomf() * MAX_WORLD_DIST;
```
Comment (jdolan) [inline src/game/default/g_item.c]: Hah!

File: src/game/default/g_item.c [hunk 3]
```diff
@@ -812,6 +820,160 @@ static void G_UseItem(g_entity_t *ent, g_entity_t *other, g_entity_t *activator)
...
+}
+
+/**
+ * @brief Drop the tech.
+ */
+static g_entity_t *G_DropTech(g_entity_t *ent, const g_item_t *item) {
+	return G_TossTech(ent);
+}
+
+/**
```
Comment (jdolan) [inline src/game/default/g_item.c]: This file is turning into a beast. Might be time to break out `g_item_tech.c`, `g_item_flag.c`, etc..

File: src/game/default/g_item.h [hunk 4]
```diff
@@ -33,10 +33,46 @@
...
+	 * @brief Strength scaling factor
+	 */
+	#define TECH_STRENGTH_DAMAGE_FACTOR 1.5
+	#define TECH_STRENGTH_KNOCKBACK_FACTOR 1.25
+
+	/**
+	 * @brief Regen constants
+	 */
+	#define TECH_REGEN_TICK_TIME 500
+	#define TECH_REGEN_HEALTH 5
```
Comment (jdolan) [inline src/game/default/g_item.h]: Whoa! Seems steep! 10 health per second?

File: src/game/default/g_main.c [hunk 5]
```diff
@@ -182,9 +183,16 @@ void G_ResetItems(void) {
 			continue;
 		}
 
+		if (ent->locals.item->type == ITEM_TECH) { // free techs, we'll spawn them again
+			G_FreeEntity(ent);
+			continue;
```
Comment (jdolan) [inline src/game/default/g_main.c]: Do we want the ability for mappers to place techs specifically, and then we only randomly spawn them when they're requested and the map does *not* include them?

File: src/game/default/g_main.c [hunk 6]
```diff
@@ -211,6 +219,28 @@ void G_CheckHook(void) {
...
+			if (g_strcmp0(g_game.spawn.techs, "default")) {
+				g_level.techs = !!atoi(g_game.spawn.techs);
+			} else {
+				g_level.techs = g_level.ctf;
+			}
+		} else {
+			g_level.techs = g_level.ctf;
+		}
+	}
+}
```
Comment (jdolan) [inline src/game/default/g_main.c]: Beautiful.

File: src/game/default/g_types.h [hunk 7]
```diff
@@ -456,6 +457,19 @@ typedef enum {
...
+ */
+typedef enum {
+	TECH_HASTE,
+	TECH_REGEN,
+	TECH_SHIELD,
+	TECH_STRENGTH,
+	TECH_VAMPIRE,
+
+	TECH_TOTAL
+} g_tech_t;
```
Comment (jdolan) [inline src/game/default/g_types.h]: Did you wire up placeholder sounds for each of these when they are applied?

File: src/game/default/g_weapon.c [hunk 8]
```diff
@@ -277,6 +277,10 @@ static void G_WeaponFired(g_entity_t *ent, uint32_t interval, uint32_t ammo_need
 	// set the attack animation
 	G_SetAnimation(ent, ANIM_TORSO_ATTACK1, true);
 
+	if (G_HasTech(ent, TECH_HASTE)) {
+		interval /= TECH_HASTE_FACTOR;
```
Comment (jdolan) [inline src/game/default/g_weapon.c]: Might be smarter to define this as 0.75 and multiply.

File: src/game/default/g_item.h (hunk 4 above)
Comment (Paril) [inline src/game/default/g_item.h]: Used Q2 values for now, we can tweak.

File: src/game/default/g_main.c (hunk 5 above)
Comment (Paril) [inline src/game/default/g_main.c]: Maybe, seems like a very specific feature but I can see it being useful (fight for the center techs).

File: src/game/default/g_types.h (hunk 7 above)
Comment (Paril) [inline src/game/default/g_types.h]: Currently no, will do that

File: src/game/default/g_cmd.c [hunk 9]
```diff
@@ -618,6 +621,7 @@ static void G_PlayerList_f(g_entity_t *ent) {
 static const char *vote_cmds[] = {
 	"g_capture_limit",
 	"g_ctf",
+	"g_techs",
```
Comment (Paril) [inline src/game/default/g_cmd.c]: Yep

**Pull request merged**